zk_prover/src/chips/overflow/overflow_check.rs

```diff
@@ -6,43 +6,48 @@ use halo2_proofs::plonk::{Advice, Column, ConstraintSystem, Error, Expression, F
 use halo2_proofs::poly::Rotation;
 use std::fmt::Debug;
 
+const MOD_BITS: usize = 252;
+
 #[derive(Debug, Clone)]
-pub struct OverflowCheckConfig<const MAX_BITS: u8, const ACC_COLS: usize> {
+pub struct OverflowCheckConfig<const MAX_BITS: u8> {
     pub a: Column<Advice>,
-    pub decomposed_values: [Column<Advice>; ACC_COLS],
+    pub b: Column<Advice>,
     pub range: Column<Fixed>,
-    pub selector: Selector,
+    pub toggle_overflow_check: Selector,
 }
 
 #[derive(Debug, Clone)]
-pub struct OverflowChip<const MAX_BITS: u8, const ACC_COLS: usize> {
-    config: OverflowCheckConfig<MAX_BITS, ACC_COLS>,
+pub struct OverflowChip<const MAX_BITS: u8> {
+    config: OverflowCheckConfig<MAX_BITS>,
 }
 
-impl<const MAX_BITS: u8, const ACC_COLS: usize> OverflowChip<MAX_BITS, ACC_COLS> {
-    pub fn construct(config: OverflowCheckConfig<MAX_BITS, ACC_COLS>) -> Self {
+impl<const MAX_BITS: u8> OverflowChip<MAX_BITS> {
+    pub fn construct(config: OverflowCheckConfig<MAX_BITS>) -> Self {
         Self { config }
     }
 
-    pub fn configure(meta: &mut ConstraintSystem<Fp>) -> OverflowCheckConfig<MAX_BITS, ACC_COLS> {
-        let selector = meta.selector();
+    pub fn configure(
+        meta: &mut ConstraintSystem<Fp>,
+        a: Column<Advice>,
+        b: Column<Advice>,
+    ) -> OverflowCheckConfig<MAX_BITS> {
         let range = meta.fixed_column();
-        let a = meta.advice_column();
-        let decomposed_values = [(); ACC_COLS].map(|_| meta.advice_column());
+        let toggle_overflow_check = meta.complex_selector();
+
+        let num_rows = MOD_BITS / MAX_BITS as usize;
```

Comment:
It may end up that num_rows is not an integer number if MAX_BITS is an odd number. I don't know if there's a situation in which makes sense to use odd `MAX_BITS`, maybe we can add an assert to avoid this.

Reply:
The num_rows is currently calculated as an integer, regardless of whether `MAX_BITS` is an odd number. |
However, I think it's better to add a warning sign for when `MAX_BITS` is an odd number, as future developers might expect different results.

Reply:
Added Warn at [ffce080](https://github.com/summa-dev/summa-solvency/pull/68/commits/ffce0801a8248d428b00d7e0c954c8a2ee794fc4)

Reply:
this commit checks whether the `MOD_BITS` was divided evenly by `MAX_BITS`